piecrust/src/session.rs

```diff
@@ -451,47 +450,35 @@ impl Session {
 
     /// Returns the current length of the memory of the given contract.
     ///
-    /// If the contract does not exist, or is otherwise not instantiated in this
-    /// session, it will return `None`.
-    pub fn memory_len(&self, contract_id: &ContractId) -> Option<usize> {
-        self.instance(contract_id)
-            .map(|instance| instance.mem_len())
+    /// If the contract does not exist, it will return `None`.
+    pub fn memory_len(
+        &mut self,
+        contract_id: ContractId,
```

Comment:
Is there a reason why this is changed to consume the `contract_id`?

Reply:
Yes. Previously it used the `WrappedInstance::mem_len` function internally, but now it uses `ContractSession::contract`. The latter takes a `ContractId` rather than a reference to it. It's essentially for the same reason as why we changed the return to be a `Result`.

This is kk though, since `ContractId` is pretty small and `Copy`.